internal/services/azurestackhci/stack_hci_marketplace_gallery_image_resource_test.go

```diff
@@ -0,0 +1,283 @@
+package azurestackhci_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/azurestackhci/2024-01-01/marketplacegalleryimages"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type StackHCIMarketplaceGalleryImageResource struct {
+	// az vm image list --all --output table --sku 2022-datacenter-azure-edition-core
+	imageVersion string
+}
+
+func TestAccStackHCIMarketplaceGalleryImage_basic(t *testing.T) {
+	if os.Getenv(customLocationIdEnv) == "" {
+		t.Skipf("skipping since %q has not been specified", customLocationIdEnv)
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_stack_hci_marketplace_gallery_image", "test")
+	r := StackHCIMarketplaceGalleryImageResource{
+		imageVersion: "20348.2402.240607",
+	}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccStackHCIMarketplaceGalleryImage_complete(t *testing.T) {
+	if os.Getenv(customLocationIdEnv) == "" {
+		t.Skipf("skipping since %q has not been specified", customLocationIdEnv)
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_stack_hci_marketplace_gallery_image", "test")
+	r := StackHCIMarketplaceGalleryImageResource{
+		imageVersion: "20348.2582.240703",
+	}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccStackHCIMarketplaceGalleryImage_update(t *testing.T) {
+	if os.Getenv(customLocationIdEnv) == "" {
+		t.Skipf("skipping since %q has not been specified", customLocationIdEnv)
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_stack_hci_marketplace_gallery_image", "test")
+	r := StackHCIMarketplaceGalleryImageResource{
+		imageVersion: "20348.2655.240810",
+	}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
```

Comment:
it only test updating `tags`, please also test adding/removing `tags`.

Reply:
fixed